mifosng-android/src/main/java/com/mifos/mifosxdroid/online/generatecollectionsheet/GenerateCollectionSheetFragment.java

```diff
@@ -128,181 +197,670 @@ public void inflateGroupSpinner(final int officeId, int staffId) {
-                android.R.layout.simple_spinner_item, staffNames);
-
-        staffAdapter.notifyDataSetChanged();
-        staffAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        sp_loan_officers.setAdapter(staffAdapter);
-        sp_loan_officers.setOnItemSelectedListener(
-                new AdapterView.OnItemSelectedListener() {
-                    @Override
-                    public void onItemSelected(AdapterView<?> parent, View view, int
-                            position,
-                                               long id) {
-                        int staffId = staffNameIdHashMap.get(staffNames.get(position));
-                        if (staffId != -1) {
-                            inflateCenterSpinner(officeId, staffId);
-                            inflateGroupSpinner(officeId, staffId);
-                        } else {
-                            Toaster.show(rootView, getString(R.string.error_select_staff));
-                        }
+    private void fetchProductiveCollectionSheet() {
+        //Make RequestPayload and retrieve Productive CollectionSheet.
+        CollectionSheetRequestPayload requestPayload = new CollectionSheetRequestPayload();
+        requestPayload.setTransactionDate(tvMeetingDate.getText().toString());
+        requestPayload.setCalendarId(calendarId);
+        presenter.loadProductiveCollectionSheet(productiveCenterId, requestPayload);
 
-                    }
+    }
 
-                    @Override
-                    public void onNothingSelected(AdapterView<?> parent) {
+    private void fetchCenterDetails() {
+        presenter.loadCenterDetails("dd MMMM yyyy", "en",
+                tvMeetingDate.getText().toString(), officeId, staffId);
+    }
 
-                    }
-                });
+    @Override
+    public void onCenterLoadSuccess(List<CenterDetail> centerDetails) {
+        //TODO Check implementation if List contains more than one centerDetail
+        //Set CalendarId and fetch ProductiveCollectionSheet
+        calendarId = centerDetails.get(0).getMeetingFallCenters().get(0).
+                getCollectionMeetingCalendar().getId();
+        productiveCenterId = centerDetails.get(0).getMeetingFallCenters().get(0).getId();
+        fetchProductiveCollectionSheet();
     }
 
     @Override
-    public void showCentersInOffice(List<Center> centers) {
-        final List<String> centerNames = new ArrayList<String>();
+    public void showProductive(CollectionSheetResponse sheet) {
+        inflateProductiveCollectionTable(sheet);
+    }
+
+    @Override
+    public void showCollection(CollectionSheetResponse sheet) {
+        inflateCollectionTable(sheet);
+    }
 
-        centerNames.add(getString(R.string.spinner_center));
-        centerNameIdHashMap.put(getString(R.string.spinner_center), -1);
+    private void inflateCollectionTable(CollectionSheetResponse s) {
```

Comment:
using `s` It's not a good naming convention make the change to `collectionSheetResponse`.